visma/solvers/simulEqn.py

```diff
@@ -0,0 +1,132 @@
+
+    Returns:
+        coefficients -- 3 X 4 list -- each each row contains coefficients for x, y, z and constant term respectively.
+    '''
+    # TODO: Currently implemented taking with fixed variables x, y and z. To be implemented with different variables like u, v  and w.
+    coefficients = []
+    coefficients = [[0] * 4 for _ in range(3)]
+    for i, LandR_token in enumerate(LandR_tokens):
+        lTokens = LandR_token[0]
+        rTokens = LandR_token[1]
+        if len(rTokens) > 0:
+            lTokens, rTokens = moveRTokensToLTokens(
+                lTokens, rTokens
+            )
+        for _, token in enumerate(lTokens):
+            if isinstance(token, Variable):
+                if token.value == ['x']:
+                    coefficients[i][0] = token.coefficient
+                elif token.value == ['y']:
+                    coefficients[i][1] = token.coefficient
+                elif token.value == ['z']:
+                    coefficients[i][2] = token.coefficient
+            if isinstance(token, Constant):
+                coefficients[i][3] = token.value
+
+    return coefficients
+
+
+def crammerMatrices(coefficients):
+    '''
+    Arguments:
+    coefficients -- 3 X 4 list -- each each row contains coefficients for x, y, z and constant term respectively
+
+    Returns:
+    Dx, Dy, Dz, D -- 3 X 4 list -- Crammer's Matrices for implementing Crammer's Rule.
+    '''
+    D = [[0] * 3 for _ in range(3)]
+    Dx = [[0] * 3 for _ in range(3)]
+    Dy = [[0] * 3 for _ in range(3)]
+    Dz = [[0] * 3 for _ in range(3)]
+    for i in range(3):
+        for j in range(3):
+            D[i][j] = coefficients[i][j]
+            Dx[i][j] = coefficients[i][j]
+            Dy[i][j] = coefficients[i][j]
+            Dz[i][j] = coefficients[i][j]
+    for k in range(3):
+        Dx[k][0] = coefficients[k][3]
+        Dy[k][1] = coefficients[k][3]
+        Dz[k][2] = coefficients[k][3]
+    for i in range(3):
+        for j in range(3):
+            D[i][j] = tokenizer(str(D[i][j]))
+            Dx[i][j] = tokenizer(str(Dx[i][j]))
+            Dy[i][j] = tokenizer(str(Dy[i][j]))
+            Dz[i][j] = tokenizer(str(Dz[i][j]))
+    matD = SquareMat()
+    matD.value = D
+    matDx = SquareMat()
+    matDx.value = Dx
+    matDy = SquareMat()
+    matDy.value = Dy
+    matDz = SquareMat()
+    matDz.value = Dz
+    return matD, matDx, matDy, matDz
+
+
+def getResult(matD, matDx, matDy, matDz):
+    '''Calculates values of x, y and z'''
+    x = 0
+    y = 0
+    z = 0
+    trivial = True
+    detD = float(tokensToString(matD.determinant()))
+    detDx = float(tokensToString(matDx.determinant()))
+    detDy = float(tokensToString(matDy.determinant()))
+    detDz = float(tokensToString(matDz.determinant()))
+    if detD == 0:
+        trivial = False
+    if trivial:
+        x = detDx/detD
+        y = detDy/detD
+        z = detDz/detD
+    return trivial, x, y, z
+
+
+def simulSolver(eqStr1, eqStr2, eqStr3, solveFor):
```

Comment:
When implementing with the GUI/CLI, make sure to change input arguments to type tokens instead of a string.
```suggestion
def simulSolver(eqTok1, eqTok2, eqTok3, solveFor):
```
No need to change it right now but make sure that the tokenization for this must be added in io/tokenizer.

Reply:
For now, I have added the tokenization of such input strings (_involving `;`_) in the module itself. Will modify `tokenizer.py` accordingly to incorporate such inputs (involving multiple equations) when I will make it implement it in CLI/GUI.